src/components/inputs/z-combobox/index.tsx

```diff
@@ -147,6 +138,93 @@ export class ZCombobox {
     this.selectedCounter = this.itemsList.filter((item) => item.checked).length;
   }
 
+  private getControlToListSize(): ListSize {
+    let size;
+    switch (this.size) {
+      case ControlSize.BIG:
+        size = ListSize.LARGE;
+        break;
+      case ControlSize.SMALL:
+        size = ListSize.SMALL;
+        break;
+      case ControlSize.X_SMALL:
+        size = ListSize.SMALL;
+        break;
+      default:
+        size = ListSize.MEDIUM;
+    }
+
+    return size;
+  }
+
+  private getItemId(item: ComboItem): string {
+    return `combo-checkbox-${this.inputid}-${item.id}`;
+  }
+
+  private getCheckAllId(): string {
+    return `combo-checkbox-${this.inputid}-check-all`;
+  }
+
+  private resetInputTabIndex(): void {
+    this.element.shadowRoot.querySelector(`#${this.getCheckAllId()}`).setAttribute("tabindex", "-1");
+    this.itemsList.forEach(function (item: ComboItem) {
+      this.element.shadowRoot.querySelector(`#${this.getItemId(item)}`).setAttribute("tabindex", "-1");
+    }, this);
+  }
+
+  private setInputTabIndex(itemId, index): number {
+    if (index === 0) {
+      return 0;
+    }
+
+    return itemId === this.focusedItemId ? 0 : -1;
+  }
+
+  private handleSelectArrowsNavigation(e: KeyboardEvent, key: number): void {
+    const arrows = [KeyboardCode.ARROW_DOWN, KeyboardCode.ARROW_UP];
+    if (!arrows.includes(e.key as KeyboardCode) || !this.isopen) {
+      return;
+    }
+
+    e.preventDefault();
+    e.stopPropagation();
+    let index = key;
+    if (e.key === KeyboardCode.ARROW_DOWN) {
+      index = index === this.itemsList.length - 1 ? -1 : key + 1;
```

Comment:
In più si spacca quando cerchi qualcosa e poi navighi con le frecce, non possiamo fare affidanemento su itemsList ma su renderItemsList. E poi non possiamo ragionare per indice perché l'indice cambia quando cerchi.